modules/openapi-generator/src/main/resources/elixir/request_builder.ex.mustache

```diff
@@ -107,20 +107,22 @@ defmodule {{moduleName}}.RequestBuilder do
 
   ## Parameters
 
-  - arg1 (Tesla.Env.t | term) - The response object
+  - arg1 ({:ok, Tesla.Env.t} | term) - The response object
   - arg2 (:false | struct | [struct]) - The shape of the struct to deserialize into
 
   ## Returns
 
   {:ok, struct} on success
   {:error, term} on failure
   """
-  @spec decode(Tesla.Env.t | term()) :: {:ok, struct()} | {:error, Tesla.Env.t} | {:error, term()}
-  def decode(%Tesla.Env{status: 200, body: body}), do: Poison.decode(body)
+  @spec decode({:ok, Tesla.Env.t} | term()) :: {:ok, struct()} | {:error, Tesla.Env.t} | {:error, term()}
+  def decode({:ok, %Tesla.Env{status: 200, body: body}}), do: Poison.decode(body)
   def decode(response), do: {:error, response}
+  def decode({:error, _} = error), do: error
```

Comment:
Since `def decode(response)` would match everything it needs to the last function signature for `decode/1`

```suggestion
  def decode({:error, _} = error), do: error
  def decode(response), do: {:error, response}
```